Autopilot/PathManager/Src/pathStateClasses.cpp

```diff
@@ -120,4 +324,377 @@ pathManagerState& fatalFailureMode::getInstance()
+    output = LandingManager::translateWaypointCommands(waypointOutput);
+
+    //calculating the difference in heading to detect if finished turning (2 differences possible)
+    differenceInHeading1 = input.telemetryData.stoppingDirectionHeading - input.sensorOutput.track;
+    differenceInHeading2 = input.sensorOutput.track - input.telemetryData.stoppingDirectionHeading;
+
+    //making sure both headings are positive
+    if(differenceInHeading1 < 0){differenceInHeading1 += 360;}
+    if(differenceInHeading2 < 0){differenceInHeading2 += 360;}
+
+    //if the smaller heading is less than 5 degrees, set stage to slope
+    if((differenceInHeading1 < differenceInHeading2 && fabs(differenceInHeading1) <= 5) || (fabs(differenceInHeading2) <= 5))
+    {
+        pathMgr->stage = SLOPE;
+    }
+
+    if(!landingTransitionStage::waypointStatus == WAYPOINT_SUCCESS)
+    {
+        pathMgr->isError = true;
+    }
+    
+    if(pathMgr->isError)
+    {
+        pathMgr->setState(fatalFailureMode::getInstance());
+    }
+    else
+    {
+        pathMgr->setState(coordinateTurnElevation::getInstance());
+    }
+}
+
+pathManagerState& landingTransitionStage::getInstance()
+{
+    static landingTransitionStage singleton;
+    return singleton;
+}
+
+void landingSlopeStage::execute(pathManager* pathMgr)
+{
+    //load in sensor fusion data and telemtry data into input structure
+    input.telemetryData = *(commsWithTelemetry::GetTelemetryIncomingData());
+    input.sensorOutput = *(sensorFusion::GetSFOutput());
+
+    if(input.sensorOutput.altitude <= (FLARE_ALTITUDE + input.telemetryData.stoppingAltitude)) //if less than flare altitude
+    {
+        pathMgr->stage = FLARE;
+
+    }
+    else
+    {
+        //setting sensorFusion input to waypoint data in
+        waypointInput.latitude = input.sensorOutput.latitude;
+        waypointInput.longitude = input.sensorOutput.longitude;
+        waypointInput.altitude = input.sensorOutput.altitude;
+        waypointInput.track = input.sensorOutput.track;
+
+        //aligning horizontal position using waypointManager get_next_directions
+        landingTransitionStage::waypointStatus = landingTransitionStage::landingPath.get_next_directions(waypointInput, &waypointOutput);
+        
+        //translate waypoint ouput data into landing data
+        output = LandingManager::translateWaypointCommands(waypointOutput);
+        
+        //retrieving desired altitude for slope state and setting it 
+        output.desiredAltitude = LandingManager::changingAltitude(input.sensorOutput,landingTransitionStage::path.aimingPoint, landingTransitionStage::path.intersectionPoint, landingTransitionStage::path.stoppingPoint);
+        
+        //retrieving desired speed for approach speed and setting it
+        output.desiredAirspeed = LandingManager::approachSpeed(pathMgr->isPackage);
+    }
+    
+    if(!landingTransitionStage::waypointStatus == WAYPOINT_SUCCESS)
+    {
+        pathMgr->isError = true;
+    }
```

Comment:
Should be `landingTransitionStage::waypointStatus != WAYPOINT_SUCCESS` since it's an integer and not a boolean.

Reply:
I think you may have done this in other places too

Reply:
oops, messed up the precedences. Will fix